From c093aff6c4a5c75a387efd90ccae734412ff7388 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 22 Aug 2022 15:23:35 +0800
Subject: [PATCH] [riscv64][wasm-relaxed-simd] Implement relaxed
 i16x8.q15mulr_s

Port commit aaa15e65d40d8af4933d3788c5855f1205297ed4

Change-Id: I728d5f786c3e217d249bf6d356b2a004896ed582
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3844663
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
Reviewed-by: Tobias Tebbi <tebbi@chromium.org>
Commit-Queue: Tobias Tebbi <tebbi@chromium.org>
Cr-Commit-Position: refs/heads/main@{#82617}
---
 src/compiler/backend/instruction-selector.cc            | 2 +-
 src/compiler/backend/riscv/instruction-selector-riscv.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/compiler/backend/instruction-selector.cc b/src/compiler/backend/instruction-selector.cc
index e963132d634..a6dd6d86898 100644
--- a/src/compiler/backend/instruction-selector.cc
+++ b/src/compiler/backend/instruction-selector.cc
@@ -2837,7 +2837,7 @@ void InstructionSelector::VisitI32x4RelaxedTruncF32x4U(Node* node) {
         // !V8_TARGET_ARCH_RISCV32
 
 #if !V8_TARGET_ARCH_ARM64 && !V8_TARGET_ARCH_ARM && !V8_TARGET_ARCH_X64 && \
-    !V8_TARGET_ARCH_IA32
+    !V8_TARGET_ARCH_IA32 && !V8_TARGET_ARCH_RISCV64 && !V8_TARGET_ARCH_RISCV32
 void InstructionSelector::VisitI16x8RelaxedQ15MulRS(Node* node) {
   UNIMPLEMENTED();
 }
diff --git a/src/compiler/backend/riscv/instruction-selector-riscv.h b/src/compiler/backend/riscv/instruction-selector-riscv.h
index 1ac81a26995..42cae64f214 100644
--- a/src/compiler/backend/riscv/instruction-selector-riscv.h
+++ b/src/compiler/backend/riscv/instruction-selector-riscv.h
@@ -944,6 +944,7 @@ void InstructionSelector::VisitI16x8ExtAddPairwiseI8x16U(Node* node) {
   V(I16x8GeU, kRiscvI16x8GeU)                           \
   V(I16x8RoundingAverageU, kRiscvI16x8RoundingAverageU) \
   V(I16x8Q15MulRSatS, kRiscvI16x8Q15MulRSatS)           \
+  V(I16x8RelaxedQ15MulRS, kRiscvI16x8Q15MulRSatS)       \
   V(I16x8SConvertI32x4, kRiscvI16x8SConvertI32x4)       \
   V(I16x8UConvertI32x4, kRiscvI16x8UConvertI32x4)       \
   V(I8x16Add, kRiscvI8x16Add)                           \
-- 
2.35.1

